shop/src/main/java/ro/msg/learning/shop/controller/CustomerController.java

```diff
@@ -0,0 +1,63 @@
+package ro.msg.learning.shop.controller;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import ro.msg.learning.shop.dto.CustomerDto;
+import ro.msg.learning.shop.dto.SupplierDto;
+import ro.msg.learning.shop.service.CustomerService;
+import ro.msg.learning.shop.service.exceptions.ProductException;
+
+import javax.servlet.http.HttpServletResponse;
+import java.util.List;
+
+@RestController
+public class CustomerController {
+    private final CustomerService customerService;
+
+    public CustomerController(CustomerService customerService)
+    {
+        this.customerService = customerService;
+    }
+
+    @GetMapping("/customers")
+    @ResponseBody
+    public List<CustomerDto> getAllCustomers(){
+        return customerService.getAllCustomers();
+    }
+
+    @GetMapping("/customers/{customerId}")
+    public ResponseEntity<CustomerDto> findCustomerById(@PathVariable Long customerId)
+    {
+        try{
+            CustomerDto customerDto = customerService.findCustomerById(customerId);
+            return ResponseEntity.status(HttpStatus.OK).body(customerDto);
+        }catch (ProductException ex)
+        {
+            return ResponseEntity.status(HttpStatus.NOT_FOUND).body(null);
+        }
+    }
+
+    @PostMapping("/customers")
+    public CustomerDto addCustomer(@RequestBody CustomerDto customerDto, HttpServletResponse response)
+    {
+        return customerService.addCustomer(customerDto);
+    }
+
+    @PutMapping("/customers/{customerId}")
+    public void updateSupplier(@PathVariable Long customerId,@RequestBody CustomerDto customerDto) throws ProductException {
+        customerService.updateCustomer(customerId,customerDto);
```

Comment:
As I suggested in the daily also, we shouldn't use the Dtos in the service layer. 
So here you should do the mapping to the model and call the service with the model, and not the Dto.
Do the same changes for all of your services

Reply:
The controller can't create domain objects because it doesn't have direct access to the repositories needed to create them. This is especially true for entities involved in one-to-many relationships. That's why when we want to create an entity from the domain, we communicate the necessary information to the service through a dto. The service will return the entity which can be mapped to a dto in the controller and then be returned.

Reply:
You don't have to create domain objects in the controller, you just map the dto from the request to the model (filling just the data you have, not necessary all of them. The service will take care to fill all the data by making retrieve request to the repository). It's not a good practice to pass Dto to the Service and return entity from the Service. 

The reason for not using Dto in the Service is that you should be able to use that service in any application, independent of the Dto used (each application defines it's own Dto).